typeclasses/chargen.py

```diff
@@ -0,0 +1,587 @@
+                                                             reset=True),
+                "goto": "menunode_allocate_traits"},
+               {"key": ("No", "n", "_default"),
+                "desc": "Return to Archetype selection",
+                "goto": "menunode_welcome_archetypes"})
+    return text, options
+
+
+def menunode_allocate_traits(caller, raw_input):
+    """Discretionary trait point allocation menu node."""
+    char = caller.new_char
+    text = ""
+    if raw_input.isdigit() and int(raw_input) <= len(archetypes.PRIMARY_TRAITS):
+        chartrait = char.traits[archetypes.PRIMARY_TRAITS[int(raw_input)-1]]
+        if chartrait.actual < 10:
+            chartrait.mod += 1
+        else:
+            text += "|rCannot allocate more than 10 points to one trait!|n\n"
+
+    data = []
+    for i in xrange(3):
+        data.append([_format_trait_3col(char.traits[t])
+                     for t in archetypes.PRIMARY_TRAITS[i::3]])
+    table = EvTable(header=False, table=data)
+    remaining = archetypes.get_remaining_allocation(char.traits)
+
+    text += "Your character's traits influence combat abilities and skills.\n"
+    text += "Type 'help' to see individual skill definitions.\n\n"
+    text += "Allocate additional trait points as you choose.\n"
+    text += "Current:\n{}".format(table)
+    text += "\n  |w{}|n Points Remaining\n".format(remaining)
+    text += "Please select a trait to increase:"
+
+    help = "Traits:\n"
+    help += "  Strength - affects melee attacks, fortitude saves, and physical tasks\n"
+    help += "  Perception - affects ranged attacks, reflex saves, and perception tasks\n"
+    help += "  Intelligence - affects skill bonuses, will saves, and magic\n"
+    help += "  Dexterity - affects unarmed attacks, defense, and dexterity-based skills\n"
+    help += "  Charisma - affects skills related to interaction with others\n"
+    help += "  Vitality - affects health and stamina points"
+
+    options = [{"desc": char.traits[t].name,
+                "goto": "menunode_allocate_traits"}
+               for t in archetypes.PRIMARY_TRAITS]
+    options.append({"desc": "Start Over",
+                    "exec": lambda s: archetypes.apply_archetype(
+                                          s.new_char,
+                                          s.new_char.db.archetype,
+                                          reset=True),
+                    "goto": "menunode_allocate_traits"})
+
+    if remaining > 0:
+        return (text, help), options
+    else:
+        return menunode_races(caller, "Final Skills:\n{}".format(table))
+
+
+def menunode_races(caller, raw_input):
+    """Race listing menu node."""
+    text = ""
+    if raw_input and raw_input[0] == 'F':
+        text += raw_input
+
+    text += "\n\nNext, select a race for your character."
+
+    help = fill("Your race gives your character certain bonuses and"
+                "detriments, and makes certain 'focuses' available. "
```

Comment:
Yeah, there is. I'm on the fence about using `fill` in general here. In the longer menu nodes, I tended to manually format the line breaks.